src/pyupgrade/plugins/rewrite_mock_import.rs

```diff
@@ -0,0 +1,150 @@
+use rustpython_ast::{AliasData, Located, Stmt, StmtKind};
+
+use crate::ast::types::Range;
+use crate::ast::whitespace::indentation;
+use crate::autofix::Fix;
+use crate::checkers::ast::Checker;
+use crate::checks::{Check, CheckKind};
+use crate::isort::helpers::trailing_comma;
+use crate::isort::types::TrailingComma;
+
+/// Replaces a given statement with a string
+fn update_import(checker: &mut Checker, stmt: &Stmt, new_stmt: String) {
+    let mut check = Check::new(CheckKind::RewriteMockImport, Range::from_located(stmt));
+    if checker.patch(check.kind.code()) {
+        check.amend(Fix::replacement(
+            new_stmt,
+            stmt.location,
+            stmt.end_location.unwrap(),
+        ));
+    }
+    checker.add_check(check);
+}
+
+/// Returns the amount of new lines in a given statement
+fn new_line_count(checker: &Checker, stmt: &Stmt) -> usize {
+    let contents = checker
+        .locator
+        .slice_source_code_range(&Range::from_located(stmt));
+    let mut count: usize = 0;
+    // I tried using tokens here, but the count didn't work
+    for character in contents.as_ref().chars() {
+        if character == '\n' {
+            count += 1;
+        }
+    }
+    count
+}
+
+/// Create the new string for the import
+fn create_new_statement(
```

Comment:
Have you looked into using LibCST at all for these sorts of complicated rewrites? We have a few examples, like in `src/flake8_comprehensions/fixes.rs`. It gives you a concrete syntax tree (which includes punctuation, whitespace, etc.), which you can then edit, and generate fixed code from the edited tree. It might give us a lot more safety as these transforms are kind of bespoke, and I worry that they might get tripped up since we're doing so much exact-string processing.

Are you open to seeing if LibCST would be useable here?